hash_practice/exercises.py

```diff
@@ -2,20 +2,50 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    pass
+
+    dict = {}
+    for string in strings:
+        sorted_string = ''.join(sorted(string))
+        if sorted_string in dict:
+                dict[sorted_string].append(string)
+        else:
+            dict[sorted_string] = [string]
+    return_list = []
+    for key, value in dict.items():
+        return_list.append(value)
+    return return_list
+
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n log n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 However your time complexity is O(nk) due to the loop and `max` function call in lines 40-41